axiom/optimizer/connectors/hive/LocalHiveConnectorMetadata.cpp

```diff
@@ -277,9 +277,9 @@ std::pair<int64_t, int64_t> LocalHiveTableLayout::sample(
 
   const auto outputType = ROW(std::move(names), std::move(types));
 
-  auto connectorQueryCtx =
-      reinterpret_cast<LocalHiveConnectorMetadata*>(connector()->metadata())
-          ->connectorQueryCtx();
+  auto connectorQueryCtx = reinterpret_cast<LocalHiveConnectorMetadata*>(
```

Comment:
```suggestion
  auto connectorQueryCtx = static_cast<LocalHiveConnectorMetadata*>(
```
It's wrong logically and can be wrong in practice if sometime `ConnectorMetadata` won't be first base class of `LocalHiveConnectorMetadata`